### ISSUE TITLE
testing tool to verify/ensure v14 helm charts work across specified k8s releases

### ISSUE THREAD

**Issue by tdaly61** [mojaloop/project#2550]
## **Goal**:
`As a` <contributor>

`I want to` <ensure that the mojaloop v14 charts deploy and work (at a minimum) across version 1.20,1.21 and 1.22 of kubernetes. To accomplish this for v14.x.x and future releases I want to create a tool that will deploy and test the v14 charts automatically across multiple k8s releases, reporting success or failure for each release>
  
`so that` <

1. We can ensure the quality of the mojaloop charts and code across multiple releases
2. So we can also test across older releases automatically and quantify support costs and better understand and support existing deployments operational requirements
3. We streamline the v14 changes to the helm charts i.e. make sure that the helm charts design we adopt for a small number of charts is well tested against the 3 k8s releases , thus avoiding lots of post-hoc changes for many charts to make then work across these releases.  
> 

**Acceptance Criteria**:
- [ ] tool works across a number of unix based platforms 
- [ ] tool works for on-prem / laptop based testing as well as on-cloud testing
- [x] initially automatically verifies that helm chart deployment succeeds across multiple releases
- [x] initially automatically verifies that where available endpoint "health checks" pass 
- [ ] finally verifies deployment of charts (when fully converted for ML v14.x.x) by automatically running the GoldenPath tests of the TTK.

**Complexity**: <Medium>  not difficult to solve 95% of problem, challenges come from running this continually , perhaps in parallel and on-cloud

**Uncertainty**: <Low>  there are not a lot of unknowns, this is a testing tool based on much existing work e.g the TTK, mini-loop , helm etc. 

---

**Tasks**:
- [x] create tool (script) to accept parameters for k8s and ingress versions to install 
- [x] based on params automatically (and serially) deploy the existing v14 PoC charts (charts repo) into these k8s versions and report pass or fail based on successful helm deploy(s) and of functioning health check endpoints
- [ ] when v14.x.x charts are functional => add in the running of the GoldenPath tests using TTK
- [ ]  consider moving to cloud and making the tests parallel

**Done**
- [ ] Acceptance Criteria pass
- [ ] Designs are up-to date
- [ ] Unit Tests pass
- [ ] Integration Tests pass
- [ ] Code Style & Coverage meets standards
- [ ] Changes made to config (default.json) are broadcast to team and follow-up tasks added to update helm charts and other deployment config.
- [ ] TBD

## **Pull Requests**:
- [ ] TBD

## **Follow-up**:
- N/A

**Dependencies**:
- charts release v14.x.x 

## **Accountability**:
- Owner: @tdaly61 
- QA/Review: TBC

**Comment by elnyry-sam-k** [comment]
outdated, please reopen or create new issue if work on this resumes
